src/builder.rs

```diff
@@ -318,12 +348,8 @@ impl SaplingBuilder {
     ///
     /// This may be larger than the number of outputs that have been added to the builder,
     /// depending on whether padding is going to be applied.
-    pub fn bundle_output_count(&self) -> usize {
-        // This matches the padding behaviour in `Self::build`.
-        match self.spends.len() {
-            0 => self.outputs.len(),
-            _ => std::cmp::max(MIN_SHIELDED_OUTPUTS, self.outputs.len()),
-        }
+    pub fn bundle_output_count(&self, padding_rule: &PaddingRule) -> usize {
+        padding_rule.num_outputs(self.outputs.len())
```

Comment:
This now only reflects the padding rule, and not the fact that the padding is still in `build` conditional on there being real spends.

Reply:
Hm... this comment is still relevant. I think that we should remove this method entirely.